benchmarks/benchmarker.py

```diff
@@ -0,0 +1,329 @@
+import numpy as np
+import git
+import json
+import os
+import argparse
+import sys
+import unittest
+import importlib
+
+
+# To add a new benchmark,
+# 1) Make a file with name starting with benchmark_
+# 2) Inherit Benchmark class
+# 3) Implement run(on_circleci) method in the file
+
+# filename with directory to save benchmark report
+def get_report_filename(filename):
+    this_script_path = os.path.dirname(os.path.abspath(__file__))
+    report_filename = os.path.join(this_script_path, "reports", filename)
+    return report_filename
+
+
+# check if module exists and return the imported module if it exists
+def try_importing(module):
+    try:
+        return importlib.import_module(module)
+    except ImportError:
+        print("{} does not exist and thus cannot"
+              " be benchmarked".format(module))
+        return None
+
+
+# description string for benchmark run
+def benchmark_desc(name, params):
+    s = name + ": \n\t"
+    s += ", ".join("{} = {}".format(str(k), str(v)) for k, v in params.items())
+    return s
+
+
+# param name: Name of benchmark run to print
+# param Ns: List containing N values to run benchmark on
+# param number: Number of times to run time
+# param classobj: Benchmark class object to benchmark on
+# param print_stats: Print stats if true
+# param on_circleci: Limit thread number if ran on circle.ci
+# param kwargs: Initializer variables for classobj
+# return: Dictionary containing benchmark information
+def do_some_benchmarks(name, Ns, number, classobj, print_stats,
+                       on_circleci, **kwargs):
```

Comment:
In `freud`, we use Google-style docstrings (put the function documentation below the `def` and use the format below).
https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

Reply:
I'd like to rename this function to something like `run_benchmarks`.